interview-questions.md

```diff
@@ -8,48 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Researched answer: Yarn is necessary to create a React app due to the fact that a React Application is made of thousands of individual code snippets dependencies called node modules. The node modules must be organized in a particular order, and yarn ensures that they are correctly place for the framework of a React app.
 
 3. What is an anonymous function in JavaScript?
 
-Your answer:
+Your answer: An anonymous function is a function that does not have a specified parameter for what type of argument can be passed through. This allows multiple types of data to be passed through a single function for continuous iteration for different scenarios. 
 
-Researched answer:
+Researched answer: An anonymous function is a function that does not have any declared name for the function. Instead of using a function name to declare a function, ie. const functionName = () => {}, we can instead use function = () => {}. The function will only be called by a variable it is stored in as a function as a value. 
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: Props or properties are React tools that allow information to flow from one component(child) to component(child). State values allow us to use the properties from the component files inside the HTML script of the React App file. Props are specific to the components while state values are specific to the HTML script in the App folder.
 
-Researched answer:
+Researched answer: State is a variable that stores data that determines how a component will be rendered as well as its behavior. If a component contains a data containing state variable, it is called a logic operator. If it does not, then it is called a display component. Props allows for the interconnectivity between components but do not require useState() syntax or a state variable in order to function.
 
 5. What is the DOM?
 
-Your answer:
+Your answer:Document Object Model (DOM) is a visual representation of HTML code. A website in a browser is a DOM. It allows for the smooth representation of a file or app and works within React allowing immediate feedback on any changes that occur during the coding process.
 
-Researched answer:
+Researched answer: DOM essentially allows for the ability to get, change, add, or delete different HTML elements.
 
 6. STRETCH: Which is the difference between a div and a span?
 
-Your answer:
+Your answer: I honestly have no clue what the difference is between a div and a span. I know that divs allows us to essentially make containers for different objects within the HTML file. It lets us manipulate the data within the individual divs without affecting any related information within the rest of the file.
```

Comment:
Nice work here!  This is a wonderful response when you are not sure exactly how to answer the question!  👏